src/parser/reader.rs

```diff
@@ -29,7 +29,9 @@ impl<'l> Reader<'l> {
         if !block(self) {
             return None;
         }
-        let content = &self.content[start..self.offset].trim();
+        let start = self.content.char_indices().nth(start).unwrap().0;
```

Comment:
Probably it makes sense to do it on line 28. What do you think?

Reply:
Unfortunately, both putting this on line 28, or pulling line 28 here, results in failing `cargo test`s.
I think we can't pull line 28 down, because `block(&mut self)` changes `self.offset`, and we need to remember the offset before passing self into block,
And we can't do this on line 28, probably because `block` may fail and need to be returned early, and one such reason is that we're past eof (?), aka there's no character at index `start`, so the `unwrap()` in the `let start = self.content.char_indices().nth(start).unwrap().0;` panics. We can only unwrap if the `block` succeeds.